common/io.hpp

```diff
@@ -74,8 +74,7 @@ class StreamReader : virtual public Reader {
 		is_ {stream} {
 	}
 	ExpectedSize Read(vector<uint8_t> &dst) override {
-		is_.read(reinterpret_cast<char *>(&dst[0]), dst.size());
-		return is_.gcount();
+		return is_.readsome(reinterpret_cast<char *>(&dst[0]), dst.size());
```

Comment:
The documentation for this function is a bit unclear, it seems to indicate that it might return nothing if the underlying buffer happens not to be filled. And that the function will not itself fill the buffer?

I don't know, [the documentation](https://en.cppreference.com/w/cpp/io/basic_istream/readsome) is difficult to understand. What do you think?

Reply:
I think we can stay with 'read', the documentation for 'readsome' says:
"The behavior of this function is highly implementation-specific."
So it probably safer to use 'read', especially if we don't expect large delays while extracting data from streams.